docs/ckeditor/plugin.md

```diff
@@ -0,0 +1,476 @@
+
+- register the new node type in the schema, defining [where it may be inserted](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/schema.html#defining-allowed-structures) and what [type of content it can hold](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/editing-engine.html#element-types-and-custom-data)
+- define [converters](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/conversion/custom-element-conversion.html) so that CKEditor can perform three different conversions:
+    - from input HTML input to an internal, in-memory representation
+    - from internal representation back to HTML for output
+    - from internal representation to HTML for display in the editor while the user is editing
+- implement a [Command](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/core-editor-architecture.html#commands) for inserting our custom element into the editor.
+
+After we have those things in place we pretty much just need to write some glue code to write all this together as a [CKEditor plugin](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/creating-simple-plugin.html) and that's it. There are a few quirks in how this all works, but overall it's not so so bad.
+
+To show how it works we'll go through an example plugin which adds a YouTube embed to the editor, with an embedded preview that shows the video.
+
+## Writing the Editing Plugin
+The first thing we'll start with is the _editing plugin_. This plugin will register our custom object in the editor's [schema](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/architecture/editing-engine.html#schema) and it will also define 1) how to represent the node type in the editing UI and 2) how to serialize and deserialize it. We'll also write a second plugin (below) which is only concerned with providing the ability to _insert_ a YoutubeEmbed node into the document.
+
+So let's get started!
+
+### Define a new plugin
+We'll do so by creating a new class that extends `@ckeditor/ckeditor5-core/src/plugin`:
+
+```ts
+class YoutubeEmbedEditing extends Plugin {
+  init() {}
+}
+```
+
+The `init` method will be called when the plugin is instantiated, so we'll call our methods for defining the schema and the converters from there.
+
+### Define the schema
+We need to write a bit of code to let CKEditor know about what we're trying to have it do. Because we're trying to add an embedded video player we want this to be a 'root-level' node, meaning the user shouldn't be able to insert one in the middle of a paragraph. We also need to store a bit of text inside of the node, so we can store the video ID of the video which the user is embedding.
```

Comment:
Nit, take or leave: this seems like an important concept to emphasize
```suggestion
We need to write a bit of code to let CKEditor know about what we're trying to have it do. Because we're trying to add an embedded video player we want this to be a 'root-level' node. **A 'root-level' node means a user shouldn't be able to insert one in the middle of a paragraph**. We also need to store a bit of text inside of the node, so we can store the video ID of the video which the user is embedding.
```